### DISCUSSION TITLE
VDH skips the "SaveAs" dialog, downloads to somewhere, fails with error message

### DISCUSSION THREAD

**Discussion by Maxmillium** [aclap-dev/video-downloadhelper#263] [Help and Q&A]
Your question

After choosing a video to download,
https://theaquapeace.com/video.php?aff_id=3715&subid=engpr241223midnog%20-%20Copy
and the filetype to convert to, VDH skips the dialog that allows me to tell it
where to store the downloaded file.  It then fails the download with this output:

```
o is undefined

moz-extension://80f88407-48f1-4e65-be6e-05eb3417f829/background/main.js:42
aT@moz-extension://80f88407-48f1-4e65-be6e-05eb3417f829/background/main.js:42:20605
_T@moz-extension://80f88407-48f1-4e65-be6e-05eb3417f829/background/main.js:46:12485
```

After re-installing VDH (from Firefox page) and updating my copy of the CoApp,

```
curl -sSLf https://github.com/aclap-dev/vdhcoapp/releases/latest/download/install.sh | bash
```

I STILL can't specify the destination directory - "SaveAs" doesn't work - and I now get this:

```
Convert2 error:

Output /home/admin/dwhelper/All Natural Tinnitus Marine Remedy.mp4 same as Input #0 - exiting
Error opening output file /home/admin/dwhelper/All Natural Tinnitus Marine Remedy.mp4.
Error opening output files: Invalid argument
```

Addon & CoApp version and details

VDH info:

```
version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/admin/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/admin",
  "converterBinary": "/home/admin/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}
```

**Comment by GrampaWildWilly** [comment]
I don't work for VDH.  I'm just another user, like you.  Plus I don't know squat about Linux.  Still, I have a thought about something you could try.  You're running the latest CoApp but you're not running the latest VDH.  Install the VDH beta. You can find that -- and many other useful things -- by clicking the Wiki link at the top of this page. The Wiki page shows a Table of Contents down the right side. You would do well to look at all of the chapters, but for the moment, just click the chapter on Beta versions. You should be able to take it from there.

Kind of as an aside, I'm curious why you are running such an out-of-date version of Firefox.  Is that the latest version for Linux?

**Reply by GrampaWildWilly** [reply]
You might also be specific about what target type you are using for your conversion.

**Reply by GrampaWildWilly** [reply]
Plus your post is all bold.  You know you can fix that.  Look above your post to the right.  Click the 3 dots.  Then edit your post.

**Reply by Maxmillium** [reply]
Hi Willy: Good to see you're still at this!  Been a while since I "logged in", now I find everything's on github.  First time for me and I'm finding this to be the most confusing environment in my experience.

I tried multiple times to edit the "Bold" out, but every time I click "update" it bolds itself again.  Does this site really work?

Not the latest version of Firefox for Linux - just what I have running until I can fix the bloody thing - it's a long, sordid story.

I'm not a linux guru, I just like it because it doesn't start out as the swiss cheese of the internet - virtually no holes & I don't worry about viruses.  The only thing I haven't been able to do is port a copy of Lotus123 so I have to keep a Macrocrap box running for that.

The target file type is "Re-encoded to MP4-AAC" which works in our TV set....

I'll check out those Beta versions - may be the answer.  Now I see a learning curve just navigating GitHub.

Thanks for responding....

**Reply by GrampaWildWilly** [reply]
Yeah, they can't keep me away.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/60900148-3608-4432-bcd0-7a76f62c5c67)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/18008560-fc6b-4c7a-ab73-78a8244d5128)

Even though I'm on Windows 11, I use Apache Open Office.  I believe they actually develop that on Linux.  It has the great advantage of being free.  Plus I believe they have it ported to multiple platforms, sort of like how our VDH/CoApp here works on Mac, Linux, & Windows.  That should address your Lotus issues.  Geez.  Lotus?  You really don't get out of your cave much.

You will find quickly enough that the GitHub user interface is actually far superior to what we struggled with on Google Groups.  You can actually edit your own posts here, as you've already taken advantage of . . . or tried to.  You can even change the title of a thread after the fact.  There are times when you ought to correct your spelling.  I remember one thread in the old forum that had "Chorme" in the thread title.  That would pretty much exclude it from a search results list when somebody was searching for "Chrome."  And there was no way to correct it.  So spelling is important not just because I like to pick nits.  Which I do.  But there's a better reason to get one's spelling right.  The ability to edit your own posts is a great thing for correcting spelling.  Even if it's only one letter, go ahead & fix it.  Remember, people can search here & they're most likely to try to spell things right in their search keys, so we have to cooperate by spelling things right in our posts.  The way they handle attachments here is so much nicer, too.  You can attach multiple things in a single operation as long as they're all in the same directory.  And GitHub doesn't scramble the order of attachments like Google does.  This interface is so much easier to deal with that I don't even use the E-mail interface like I used to do in the old forum.  Plus we can say Pornhub in here without fearing some bot coming along & quarantining the post.  But we still need to be circumspect when talking about that subject.  A certain amount of self-censoring is still recommended.

If you click your avatar in the upper right corner of the page, it drops down a long menu that scrolls.  You'll see an entry on there for GitHub Docs.  You might find that useful for learning the interface here.  There is also a link at the top of every page labelled Wiki.  That is kind of a user's guide for VDH.  You should get in the habit of using that as a reference.  The links to the versions of VDH & the CoApp, including the betas, are there.  You will find that @paulrouget is very active on here.  Plus he's churning out new betas all the time.  One of them recently fixed some things with conversion, so you'll definitely want to get onto the latest beta.  Fortunately, you're on Firefox, as I am, so you get earlier access to betas since most of the development here takes place on Firefox & the other browsers get updated later, after we've done our duty as guinea pigs.

Things are hopping here.  I'm glad you've managed to relocate.

**Reply by Maxmillium** [reply]
Willy:  I don't know what to do now about the bolding.  I've edited the post 4 (?) times now, done everything I know or have been advised to unbold it. Every time I save the post, it bolds again - won't STAY unbolded. The simple act of SAVING the post bolds it. 
If my copy of Libre Office did that I would be scrapping it.

**Reply by GrampaWildWilly** [reply]
Looks unbold now.  You must have gotten it right.  Unless the Unbold Fairy came along & did it for you.

**Comment by paulrouget** [comment]
Yeah, I broke SaveAs. Known problem I'm working on.

**Reply by Maxmillium** [reply]
Aha!  Will appreciate a heads-up when it's working again....
